src/main/java/com/synopsys/integration/detect/configuration/DetectConfigurationFactory.java

```diff
@@ -191,6 +195,12 @@ public PhoneHomeOptions createPhoneHomeOptions() {
         return new PhoneHomeOptions(phoneHomePassthrough);
     }
 
+    public boolean createHasSignatureScan() {
+        boolean hss = false;
+        hss = detectConfiguration.getValue(DetectProperties.DETECT_TOOLS).containsValue(DetectTool.SIGNATURE_SCAN);
+        return hss;
+    }
```

Comment:
Need to find a way to use DetectToolFilter to do this (so excludes are considered as well as includes). Consequently the code no longer belongs here. You can probably do it directly in DetectBoot (which gets the filter).

Reply:
I'm confused here... I added SIGNATURE_SCAN to the list of excluded tools and it didn't run the signature scan even for Ephemeral.  This is used to put together the blackduckdecision which in turn is passed to the tool filter creator.  It doesn't seem to ignore the excluded tools.

Reply:
Exclusions always win over inclusions, so if the user has detect.tools=SIGNATURE_SCAN and detect.tools.excluded=SIGNATURE_SCAN that should behave identically to detect.tools NOT including SIGNATURE_SCAN. I don't think that code will, will it?

Reply:
I just tried that very use case and it did NOT include it.

Reply:
OK. I guess I don't fully understand all the interactions here. Thanks.